File: Java/src/main/java/com/gildedrose/DefaultItem.java
package com.gildedrose;

public class DefaultItem extends Item {

    public DefaultItem(String name, int sellIn, int quality) {
        super(name, sellIn, quality);
    }

    @Override
    protected void updateQualityBeforeSellIn() {
        decreaseQualityIfAbove0();
    }

    @Override
    protected void updateSellIn() {
        decreaseSellIn();
    }

    @Override
    protected void updateQualityAfterSellIn() {
        if (sellIn < 0) {
            decreaseQualityIfAbove0();
        }
    }
}

File: Kotlin/src/main/kotlin/com/gildedrose/DefaultItem.kt
package com.gildedrose

class DefaultItem(override val name: String, override var sellIn: Int, override var quality: Int) : Item(name, sellIn, quality) {

    override fun updateQuality() {
        decreaseQuality()
    }

    override fun updateSellIn() {
        decreaseSellIn()
    }

    override fun updateQualityWhenExpired() {
        decreaseQuality()
    }
}